tests/console/yast2_dns_server.pm

```diff
@@ -18,75 +18,32 @@ use utils;
 use version_utils qw(is_leap is_sle);
 use y2_common 'continue_info_network_manager_default';
 use constant RETRIES => 5;
-use yast2_widget_utils 'change_service_configuration';
-
-# Test "yast2 dhcp-server" functionality
-# Ensure that all combinations of running/stopped and active/inactive
-# can be set
-
-# Assert if the dns service is running or stopped
-sub assert_running {
-    my $self    = shift;
-    my $running = shift;
-    my $cmd     = '(systemctl is-active named || true) | grep -E';
-
-    if ($running) {
-        $cmd .= ' "^active"';
-        my $counter = RETRIES;
-        # Service may take a bit of time to actually start
-        # acceptable behaviour (boo#1093029)
-        while (script_run $cmd) {
-            sleep 5;
-            $counter--;
-            # Service was not started after 25 seconds
-            die 'named service is not active even after 25s delay' if $counter == 0;
-        }
-    }
-    else {
-        $cmd .= ' "^(inactive|unknown)"';
-        record_soft_failure 'bsc#1102235' if (script_run $cmd);
-    }
-}
-
-# Assert if the dns service is enabled or disabled
-sub assert_enabled {
-    my $self    = shift;
-    my $enabled = shift;
-
-    if ($enabled) {
-        systemctl 'is-enabled named';
-    }
-    else {
-        systemctl 'is-enabled named', expect_false => 1;
-    }
-}
+use yast2_widget_utils qw(change_service_configuration verify_service_configuration);
 
 sub run {
     my $self = shift;
 
-    #
     # Preparation
-    #
+    my $older_products = is_sle('<15') || is_leap('<15.1');
+    $cmd{apply_changes} = $older_products ? "alt-a" : "alt-p";
     select_console 'root-console';
-
-    # Make sure packages are installed
-    my $firewall_package = $self->firewall;
-    zypper_call("in yast2-dns-server bind $firewall_package", timeout => 180);
-    # Let's pretend this is the first execution (could not be the case if
-    # yast2_cmdline was executed before)
-    script_run 'rm /var/lib/YaST2/dns_server';
+    zypper_call 'in yast2-dns-server bind ' . $self->firewall, timeout => 180;
+    # Pretend this is the 1st execution (could not be the case if yast2_cmdline was executed before)
+    my $dns_server_file = '/var/lib/YaST2/dns_server';
+    script_run("[[ -f $dns_server_file ]] && rm $dns_server_file");
 
     #
-    # First execution (wizard-like interface)
+    # 1st run (wizard-like interface)
     #
+    record_info '1st run',         '[wizard-like interface] -> service active & enabled';
```

Comment:
IMHO this makes the above comment obsolete. Just keep the `record_info`

Reply:
Removed all similar comments and keeping the record_info.